components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -282,61 +263,53 @@ parse_timestamp(ReaderInterface& reader, encoded_tag_t encoded_tag, epoch_time_m
 template <typename encoded_variable_t>
 static IRErrorCode
 generic_decode_next_message(ReaderInterface& reader, string& message, epoch_time_ms_t& timestamp) {
-    encoded_tag_t encoded_tag;
-    if (ErrorCode_Success != reader.try_read_numeric_value(encoded_tag)) {
-        return IRErrorCode_Incomplete_IR;
-    }
-    if (cProtocol::Eof == encoded_tag) {
-        return IRErrorCode_Eof;
-    }
+    message.clear();
 
-    // Handle variables
     vector<encoded_variable_t> encoded_vars;
     vector<string> dict_vars;
-    encoded_variable_t encoded_variable;
-    string var_str;
-    bool is_encoded_var;
-    while (is_variable_tag<encoded_variable_t>(encoded_tag, is_encoded_var)) {
-        if (is_encoded_var) {
-            if (false == decode_int(reader, encoded_variable)) {
-                return IRErrorCode_Incomplete_IR;
-            }
-            encoded_vars.push_back(encoded_variable);
-        } else {
-            if (auto error_code = parse_dictionary_var(reader, encoded_tag, var_str);
-                IRErrorCode_Success != error_code)
-            {
-                return error_code;
-            }
-            dict_vars.emplace_back(var_str);
-        }
-        if (ErrorCode_Success != reader.try_read_numeric_value(encoded_tag)) {
-            return IRErrorCode_Incomplete_IR;
-        }
-    }
-
-    // Handle logtype
     string logtype;
-    if (auto error_code = parse_logtype(reader, encoded_tag, logtype);
+    if (auto error_code
+        = generic_parse_tokens(reader, logtype, encoded_vars, dict_vars, timestamp);
         IRErrorCode_Success != error_code)
     {
         return error_code;
     }
 
-    // NOTE: for the eight-byte encoding, the timestamp is the actual timestamp;
-    // for the four-byte encoding, the timestamp is a timestamp delta
-    if (ErrorCode_Success != reader.try_read_numeric_value(encoded_tag)) {
-        return IRErrorCode_Incomplete_IR;
-    }
-    if (auto error_code = parse_timestamp<encoded_variable_t>(reader, encoded_tag, timestamp);
-        IRErrorCode_Success != error_code)
-    {
-        return error_code;
-    }
+    // constant handler
+    auto constant_handler = [&message](string const& value, size_t begin_pos, size_t length) {
+        message.append(value, begin_pos, length);
+    };
+
+    // constant handler
+    auto constant_remainder_handler = [&message](string const& value, size_t begin_pos) {
+        message.append(value, begin_pos);
```

Comment:
I got a clang-tidy warning "Calling a function that uses a default argument is disallowed". However this code style is from the previous version of code https://github.com/y-scope/clp/pull/152/files#diff-1d2bdbbd34b62a85b09c61b0baeaca2970b390837e926ad40589fb76b6b2b8c8L488. Can I ignore the warning?